### PR TITLE
ioctl supports for smart contract cmds

### PR DESCRIPTION
**Pull request by RuneAkko**
@Frankonly approve plz

### CODE REVIEW

File: ioctl/cmd/contract/contractinvokefunction.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	Args:  util.CheckArgs(1, 3),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := contractInvokeFunction(args)
+		return output.PrintError(err)
+	},
+}
+
+func contractInvokeFunction(args []string) error {
+	return nil
```
Comment (dustinxie) [inline ioctl/cmd/contract/contractinvokefunction.go:42]: no implementation?

File: ioctl/cmd/contract/contractinvokebytecode.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := contractInvokeBytecode(args)
+		return output.PrintError(err)
+	},
+}
+
+func contractInvokeBytecode(args []string) error {
+	return nil
+}
```
Comment (dustinxie) [inline ioctl/cmd/contract/contractinvokebytecode.go:42]: no implementation?

File: ioctl/cmd/contract/contractdeploysol.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	Args:  cobra.RangeArgs(1, 3),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := contractDeploySol(args)
+		return output.PrintError(err)
+	},
+}
+
+func contractDeploySol(args []string) error {
+	return nil
```
Comment (dustinxie) [inline ioctl/cmd/contract/contractdeploysol.go:41]: no implementation?

File: ioctl/cmd/contract/contractdeploybytecode.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	Args:  util.CheckArgs(1, 3),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := contractDeployBytecode(args)
+		return output.PrintError(err)
+	},
+}
+
+func contractDeployBytecode(args []string) error {
+	return nil
```
Comment (dustinxie) [inline ioctl/cmd/contract/contractdeploybytecode.go:42]: no implementation?

File: ioctl/cmd/contract/contractdeploybin.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	Args:  util.CheckArgs(1, 3),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := contractDeployBin(args)
+		return output.PrintError(err)
+	},
+}
+
+func contractDeployBin(args []string) error {
+	return nil
```
Comment (dustinxie) [inline ioctl/cmd/contract/contractdeploybin.go:42]: no implementation?
what's different Bin vs Bytecode?

Comment (RuneAkko) [reply ioctl/cmd/contract/contractdeploybin.go:42]: Yes, this is just a framework that interfaces with the compile command. details can be found  at [https://iotex.larksuite.com/docs/docusVhDwkA4lJtJhE7NFBHe5Ph](url)

**Comment by raullenchai** [comment]
Let's check in this skeleton and keep iterating.

**Pull request merged**